Makefile

```diff
@@ -303,14 +302,12 @@ clean-all: clean
 %.o: %.c $(CPYTHONLIB) $(wildcard src/core/*.h src/core/*.js)
 	$(CC) -o $@ -c $< $(MAIN_MODULE_CFLAGS) -Isrc/core/
 
-
-$(CPYTHONLIB): emsdk/emsdk/.complete
+$(CPYTHONLIB): emsdk/emsdk/.complete check-emcc
 	@date +"[%F %T] Building cpython..."
 	make -C $(CPYTHONROOT)
 	@date +"[%F %T] done building cpython..."
 
-
-dist/pyodide-lock.json: .pyodide_build_installed
+dist/pyodide-lock.json: $(CPYTHONLIB) .pyodide_build_installed
```

Comment:
This seems to [re-trigger the cpython build in our CI](https://app.circleci.com/pipelines/github/pyodide/pyodide/9733/workflows/e817b650-e7fb-4096-bff9-5ed6f413a0b0/jobs/152374), which should not, because we first build cpython, and store-reload the build artifact when building the packages.

Reply:
> Alternatively, how about moving the check-emcc logic into the emsdk/emsdk/.complete target instead?

Sounds good to me.